### PR TITLE
Fix pauseWhenLostFocus not working as expected

### PR DESCRIPTION
**Pull request by obigu**
Recently added `pauseWhenLostFocus` config was not working. First of all code was not being executed unless a `WindowListener` was added by the app and even if it was a `resume()` was missing.

### CODE REVIEW

File: backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java [hunk 1]
```diff
@@ -60,14 +60,15 @@ public void invoke (long windowHandle, final boolean focused) {
...
 						if (focused) {
 							if (config.pauseWhenLostFocus) {
 								synchronized (lifecycleListeners) {
 									for (LifecycleListener lifecycleListener : lifecycleListeners) {
 										lifecycleListener.resume();
 									}
 								}
+								listener.resume();
 							}
 							windowListener.focusGained();
```
Comment (crykn) [inline backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java:73]: This will throw an NPE if `windowListener` is null.

Comment (obigu) [reply backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java:73]: yeah, should be ok now

File: backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java [hunk 2]
```diff
@@ -60,28 +60,31 @@ public void invoke (long windowHandle, final boolean focused) {
...
+							synchronized (lifecycleListeners) {
+								for (LifecycleListener lifecycleListener : lifecycleListeners) {
+									lifecycleListener.pause();
 								}
-								listener.pause();
 							}
+							listener.pause();
 						}
-						Lwjgl3Window.this.focused = focused;
+						Lwjgl3Window.this.focused = false;
```
Comment (crykn) [inline backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java:87]: I think you accidentally moved this line into the else block. It should also stay as `Lwjgl3Window.this.focused = focused;`.

Comment (obigu) [reply backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java:87]: You're right but I'm not sure it was correct before either. Before it was only set if `windowListener != null` which doesn't make sense to me either as the `focus` property wouldn't be updated in some cases right?

Comment (crykn) [reply backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java:87]: Yes, that seems to be wrong as well.

Comment (obigu) [reply backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java]: Ok, changed, unintendedly we have fixed fix another bug in master :)

**Review by crykn — APPROVED** [review]

**Pull request merged**